### PR TITLE
Add retry-logic to sssd's join.sh script

### LINKED ISSUE #195: [ENHANCEMENT] Update `sssd` Linux domain-joiner Executor To Attempt To Retry Join Operation
Per the SaltStack Documentation, individual states support a [retry option](https://docs.saltproject.io/en/latest/ref/states/requisites.html#retrying-states). Either update the `sssd` module's `join_realm-{{ join_domain.dns_name }}` state to leverage the SaltStack `retry` option or update the `sssd` module's `join.sh` script to accept a retry-count option whose value is inherited from the formula's pillar's [`tries` parameter](https://github.com/plus3it/join-domain-formula/blob/master/pillar.example#L26).

### PR DESCRIPTION
**Pull request by ferricoxide**
Closes #195

### CODE REVIEW

File: join-domain/elx/map.jinja [hunk 1]
```diff
@@ -52,3 +52,6 @@
     'users' : admin_users + login_users,
     'logins' : admin_users + admin_groups + login_users + login_groups
 } %}
+
+# Number of times to retry domain-join operation
+{%- set tries = salt.pillar.get('join-domain:lookup:tries', 3) %}
```
Comment (eemperor) [inline join-domain/elx/map.jinja:57]: You have `tries` configured here and within the `sssd` directory.  Do you need both?

Comment (eemperor) [inline join-domain/elx/map.jinja:57]: You have `tries` configured here and within the `sssd` directory.  Do you need both?

File: join-domain/elx/sssd/map.jinja [hunk 2]
```diff
@@ -14,6 +20,7 @@ sssd_conf_parameters:
   override_homedir: '/home/%d/%u'
   shell_fallback: /bin/bash
   use_fully_qualified_names: 'False'
+tries: 3
```
Comment (eemperor) [inline join-domain/elx/sssd/map.jinja:23]: See comment above about setting a default value in two places.

File: join-domain/elx/map.jinja (hunk 1 above)
Comment (ferricoxide) [reply join-domain/elx/map.jinja:57]: When I initially added the default, here, it seemed to not actually be read (resulting in errors around the undefined value)...

File: join-domain/elx/sssd/map.jinja (hunk 2 above)
Comment (ferricoxide) [reply join-domain/elx/sssd/map.jinja:23]: ...So I added this to the formula-defaults section. I can probably remove the former, but I'll want to test it, first.

**Review by eemperor — APPROVED** [review]

**Pull request merged**